docker-compose.alfresco.yml

```diff
@@ -0,0 +1,91 @@
+version: '3.4'
+
+services:
+  db:
+    image: mdillon/postgis
+    # environment:
+    #   - POSTGRES_USER=${PG_USER:openzaak}
+    #   - POSTGRES_PASSWORD=${PG_PASSWORD:openzaak}
+
+  redis:
+    image: redis
+
+  web:
+    build: .
+    image: openzaak/open-zaak
+    environment:
+      - DJANGO_SETTINGS_MODULE=openzaak.conf.docker
+      - SECRET_KEY=${SECRET_KEY:-7(h1r2hk)8z9+05edulo_3qzymwbo&c24=)qz7+_@3&2sp=u%i}
+      - IS_HTTPS=no
+      - ALLOWED_HOSTS=*
+      - CACHE_DEFAULT=redis:6379/0
+      - CACHE_AXES=redis:6379/0
+      - SUBPATH=${SUBPATH:-/}
+      - CMIS_ENABLED=True
+      - CMIS_MAPPER_FILE=/app/config/cmis_mapper.json
+      - CMIS_BINDING=BROWSER
+    ports:
+      - 8001:8000
+    depends_on:
+      - db
+
+  nginx:
+    image: nginx
+    volumes:
+      - ./default.conf:/etc/nginx/conf.d/default.conf
+    ports:
+      - "9001:80"
+    depends_on:
+      - web
+
+  openzaak-alfresco-share:
+    image: openzaak/alfresco-share:1.0-SNAPSHOT
+    environment:
+      CATALINA_OPTS: "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=0.0.0.0:8888"
+      REPO_HOST: openzaak-alfresco-acs
+      REPO_PORT: 8080
+    ports:
+      - "11230:8080"
+      - "11231:8888"
+
+  openzaak-alfresco-acs:
+    image: openzaak/alfresco-content-services:1.0-SNAPSHOT
+    environment:
+      CATALINA_OPTS: "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=0.0.0.0:8888"
+    ports:
+      - "11232:8080"
+      - "11233:8888"
+    volumes:
+      - openzaak-alfresco-acs-volume:/usr/local/tomcat/alf_data
+    depends_on:
+      - openzaak-alfresco-postgres
+
+  openzaak-alfresco-postgres:
+    image: postgres:9.6
+    environment:
+      POSTGRES_DB: alfresco
+      POSTGRES_USER: alfresco
+      POSTGRES_PASSWORD: alfresco
+    command: postgres -c max_connections=300 -c log_min_messages=LOG
+    ports:
+      - "11234:5432"
+    volumes:
+      - openzaak-alfresco-db-volume:/var/lib/postgresql/data
```

Comment:
this should just re-use the `db` service, no? not sure why we're spinning up two postgres containers

Reply:
I won't advise putting two different datamodels in the same database. This would mean we need to create two separate databases in the same container if we only want one database container. If I want to do that, I think we would need to build a custom postgres image that allows multiple databases to be created in a single container. The default postgresql image is not setup for that.

The way it is done now, one postgresql going down due to an error does not kill database access for the other services that use the other postgresql.

Reply:
I agree, therefor we should use the initialization scripts support documented on https://hub.docker.com/_/postgres/, which can be plain SQL scripts executing the `CREATE DATABASE alfresco;` statements.

Reply:
Since I removed the Openzaak part of this docker-compose file, reusing the db service is no longer an option.

Reply:
It's definitely an option when using the `-f` flags for docker-compose - the `db` service can just be added to the `depends_on` array.

I've seen the comment about starting just Alfresco without Open Zaak and in that case, yes, you would need a DB service. This can still be achieved through the multiple compose files, but makes it a bit annoying. However, that whole set-up is unneccesary - this is the Open Zaak project which has optional support for Alfresco, not the Alfresco project with optional support for Open Zaak. Meaning - if you're starting docker-compose from Open Zaak, you are intending to run Open Zaak, always. Running Alfresco standalone is possible, but that's not something we should actively support.